README.md

```diff
@@ -128,6 +129,74 @@ with fs.scope(postcommit=True):
     fs.put("my-file.txt", "lakefs://my-repo/my-branch/my-new-file.txt")
 ```
 
+### Implicit initialization and instance caching
+
+Aside from explicit initialization, you can also use environment variables and a configuration file (by default `~/.lakectl.yaml`) to initialize a lakeFS file system.
+The environment variables for the lakeFS client arguments are the names of the constructor arguments prefixed with `LAKEFS_`:
+
+```python
+import os
+from lakefs_spec import LakeFSFileSystem
+
+os.environ["LAKEFS_HOST"] = "localhost:8000"
+os.environ["LAKEFS_USERNAME"] = "username"
+os.environ["LAKEFS_PASSWORD"] = "password"
+
+fs = LakeFSFileSystem()
+```
+
+To initialize the lakeFS file system from a `lakectl` YAML configuration file, you can specify the `configfile` argument.
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+# No argument means the default config (~/.lakectl.yaml) will be used.
+fs = LakeFSFileSystem(configfile="path/to/my/lakectl.yaml")
+```
+
+⚠️ To be able to read settings from a YAML configuration file, `pyyaml` has to be installed. You can do this by installing `lakefs-spec` together with the `yaml` extra:
```

Comment:
Should we link `PyYAML`? (LInk: https://pypi.org/project/PyYAML/)

Reply:
No, I don't think that is necessary.